src/components/InputFile/InputFile.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { FC, useState, InputHTMLAttributes } from 'react'
+import styled, { css } from 'styled-components'
+import { TextButton } from '../Button'
+import { Icon } from '../Icon'
+import { Theme, useTheme } from '../../hooks/useTheme'
+
+export type Props = InputHTMLAttributes<HTMLInputElement> & {
+  label: string
+  id?: string
+  className?: string
+  file?: File[]
+  multiple?: boolean
+}
+
+export const InputFile: FC<Props> = ({ label, id, className, file, multiple = true }) => {
+  const [files, setFiles] = useState<File[]>(file ? file : [])
+  const theme = useTheme()
+
+  const handleUpload = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files) {
+      const uploadFile = Array.from(e.target.files)
+      setFiles([...files, ...uploadFile])
+    }
+  }
+
+  const handleRemove = (index: number) => {
+    const newFiles = [...files]
+    newFiles.splice(index, 1)
+    setFiles(newFiles)
+  }
+
+  return (
+    <Wrapper className={className}>
+      {files && files.length > 0 && (
+        <FileList themes={theme}>
+          {files?.map((file, index) => {
```

Comment:
```suggestion
          {files.map((file, index) => {
```

こちらも files の存在は保証されているので optional にせずと大丈夫そうですー